### PR TITLE
feat: Add substrate message-handler

### PR DESCRIPTION
**Pull request by tcar121293**
Add substrate message-handler
## Description
Added substrate message-handler

## Related Issue Or Context

Closes: #62 

## How Has This Been Tested? Testing details.
Added unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in docs.
- [x] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: chains/substrate/executor/message-handler.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+// message handler functions for converting deposit message into a chain specific
+// proposal
+func NewSubstrateMessageHandler() *SubstrateMessageHandler {
+	return &SubstrateMessageHandler{
+		handlers: make(map[message.TransferType]MessageHandlerFunc),
+	}
+}
+
+func (mh *SubstrateMessageHandler) HandleMessage(m *message.Message) (*proposal.Proposal, error) {
+
```
Comment (mpetrun5) [inline chains/substrate/executor/message-handler.go:31]:
```suggestion
```

File: chains/substrate/executor/message-handler.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	// Based on handler that registered on BridgeContract
```
Comment (mpetrun5) [inline chains/substrate/executor/message-handler.go:32]:
```suggestion
	// Based on handler that was registered on BridgeContract
```

File: chains/substrate/executor/message-handler_test.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+//FungibleTransfer
+type FungibleTransferHandlerTestSuite struct {
+	suite.Suite
+}
+
+func TestRunFungibleTransferHandlerTestSuite(t *testing.T) {
+	suite.Run(t, new(FungibleTransferHandlerTestSuite))
+}
+
+func (s *FungibleTransferHandlerTestSuite) SetupSuite()    {}
```
Comment (mpetrun5) [inline chains/substrate/executor/message-handler_test.go:24]: Remove these if they are not used.
This is just my latest preference :smile:

File: chains/substrate/executor/message-handler_test.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+	"github.com/ChainSafe/sygma-relayer/chains/substrate/executor"
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+var errIncorrectFungibleTransferPayloadLen = errors.New("malformed payload. Len  of payload should be 2")
+var errIncorrectAmount = errors.New("wrong payload amount format")
+var errIncorrectRecipient = errors.New("wrong payload recipient format")
+
+//FungibleTransfer
```
Comment (mpetrun5) [inline chains/substrate/executor/message-handler_test.go:15]:
```suggestion
```

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

**Review by mpetrun5 — APPROVED** [review]

**Review by nmlinaric — APPROVED** [review]

**Review by MakMuftic — APPROVED** [review]

**Pull request merged**